samples/unity-of-bugs/Assets/Editor/XcodeProjectUpdater.cs

```diff
@@ -0,0 +1,42 @@
+#if UNITY_IOS && !UNITY_2021_1_OR_NEWER
+using UnityEditor;
+using UnityEditor.Callbacks;
+using UnityEditor.iOS.Xcode;
+using UnityEngine;
+
+namespace Editor
+{
+    /// <summary>
+    /// Unity adds the '-mno-thumb' compiler flag by default to iOS builds, but this flag is outdated
+    /// and no longer supported by modern versions of Xcode. Leaving this flag in the project will cause
+    /// Xcode builds to fail with an "unsupported option '-mno-thumb' for target" error. This post-process 
+    /// build step removes the flag to ensure the sample works on all versions of Unity.
+    /// </summary>
+    public static class XcodeProjectUpdater
+    {
+        [PostProcessBuild]
+        public static void OnPostProcessBuild(BuildTarget target, string pathToBuildProject)
+        {
+            if (target != BuildTarget.iOS)
+            {
+                return;
+            }
+
+            Debug.Log("XcodeUpdater.OnPostProcessBuild started.");
```

Comment:
This logger doesn't let us opt out or set level right? are we not using our logger here why?

Reply:
This is a script that lives outside the SDK and is not part of the shipped package.
It's is purely there to help us run builds when targeting iOS on old Unity versions.